src/fastoad/openmdao/problem.py

```diff
@@ -63,9 +69,23 @@ def setup(self, *args, **kwargs):
         """
         Set up the problem before run.
         """
+        problem_copy = deepcopy(self)
+        try:
+            super(FASTOADProblem, problem_copy).setup(*args, **kwargs)
+        except RuntimeError:
+            vars_metadata = self._get_undetermined_dynamic_vars_metadata(problem_copy)
+            if vars_metadata:
+                # If vars_metadata is empty, it means the RuntimeError was not because
+                # of dynamic shapes, and the incoming self.setup() will raise it.
+                ivc = om.IndepVarComp()
+                for name, meta in vars_metadata.items():
+                    ivc.add_output(name, [np.nan, np.nan], units=meta["units"])
```

Comment:
The idea behind is that if we generate an input file with no data source, I think it is better to have `[NaN, NaN]` for the dynamically-shaped inputs rather than simple `NaN`, like all other "scalar" variables.

I added a comment in ef437fc25e38a24e8e12e05c8f599b33230cf9e1.

Reply:
I get it! Thanks for adding a comment.